### PR TITLE
Switch from cypress to playwright.

### PR DESCRIPTION
**Pull request by gdborton**
**Description**
TLDR this PR:
 - [x] Switches existing integration tests from cypress to playwright
 - [x] Updates the version of yarn used to install dependencies
 - [x] Adds FF and Safari (when on mac) to the matrix of integration tests.

**Issue**
Fixes: (link to issue)

**Example**
N/A

**Context**
We're using slate for a project, but found that it has a couple of issues w/ Android (specifically around placeholders and auto-capitalize). I'm not sure I'll be able to make the change that fixes our issues, but I did want to try to contribute to the fix.

My ultimate goal here is to add an additional integration test for android that ensures autocapitalization is respected, and that the placeholder behaves as expected. To do this, I've switched the integration test runner from cypress to playwright (which has experimental android support).

As a neat side effect, the test suite is now MUCH faster. On my machine, it's come down from `7m 8s` to `1m 9s`

I've also updated the version of yarn being used as I noticed w/ the old version that CI would periodically fail.

**Checks**
- [x] The new code matches the existing patterns and styles.
- [x] The tests pass with `yarn test`.
- [x] The linter passes with `yarn lint`. (Fix errors with `yarn fix`.)
- [x] The relevant examples still work. (Run examples with `yarn start`.)
- [ ] You've [added a changeset](https://github.com/atlassian/changesets/blob/master/docs/adding-a-changeset.md) if changing functionality. (Add one with `yarn changeset add`.)

### CODE REVIEW

**Comment by bryanph** [comment]
+1 on moving to playwright, we can cover a lot more platforms that way.

Question: the integration tests say that they've run successful in the CI but when looking at the actual log output it seems a few failed?

File: playwright.config.ts [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+    use: {
+      ...devices['Desktop Safari'],
+    },
+  })
+}
+
+const retries = process.env.PLAYWRIGHT_RETRIES
+  ? +process.env.PLAYWRIGHT_RETRIES
+  : process.env.CI
+  ? 5
```
Comment (gdborton) [inline playwright.config.ts:40]: 5 retries in CI. If any test fails, it gets rerun and you get a flaky test report at the end of the run.

https://github.com/ianstormtaylor/slate/actions/runs/3859077663/jobs/6583635166#step:4:291

**Comment by bryanph** [comment]
Ah I see thanks for the clarification. Yeah I do think some of them were flaky so that sounds about right 👍
